src/transformers/models/aria/modular_aria.py

```diff
@@ -0,0 +1,1380 @@
+        tokens_per_expert (torch.Tensor): Number of tokens assigned to each expert.
+
+    Returns:
+        torch.Tensor: Output tensor of shape (num_tokens, out_features).
+    """
+    num_tokens = token_states.shape[0]
+    out_features = expert_weights.shape[-1]
+    output = torch.zeros(num_tokens, out_features, dtype=token_states.dtype, device=token_states.device)
+
+    cumsum_num_tokens = torch.cumsum(tokens_per_expert, dim=0)
+    # Insert zero at the begining for offset index's convenience
+    zero_tensor = torch.zeros(1, dtype=torch.long, device=cumsum_num_tokens.device)
+    cumsum_num_tokens = torch.cat((zero_tensor, cumsum_num_tokens))
+
+    for expert_num in range(expert_weights.shape[0]):
+        start = cumsum_num_tokens[expert_num]
+        end = cumsum_num_tokens[expert_num + 1]
+        tokens = token_states[start:end]
+
+        out = torch.matmul(tokens, expert_weights[expert_num])
+        output[start:end] = out
+    return output
+
+
+if os.environ.get("USE_GROUPED_GEMM", "1") == "0":
+    logger.warning("environment variable USE_GROUPED_GEMM is set to 0, using sequential GEMM")
+    experts_gemm = sequential_gemm
+else:
+    if importlib.util.find_spec("grouped_gemm") is None:
+        logger.warning("grouped_gemm is not installed, using sequential GEMM, which is slower.")
+        experts_gemm = sequential_gemm
+    else:
+        from grouped_gemm.ops import gmm as experts_gemm
+
+
+class AriaTextConfig(LlamaConfig):
+    """
+    Configuration class for Aria language model.
+
+    This class extends the LlamaConfig to include additional parameters specific to the Mixture of Experts (MoE) architecture.
+
+    Args:
+        moe_intermediate_size (`int`, *optional*, defaults to 4096):
+            The intermediate size for MoE layers.
+        moe_num_experts (`int`, *optional*, defaults to 8):
+            The number of experts in the MoE layer.
+        moe_topk (`int`, *optional*, defaults to 2):
+            The number of top experts to route to for each token.
+        moe_z_loss_coeff (`float`, *optional*, defaults to 1e-05):
+            The coefficient for the auxiliary z-loss.
+        moe_aux_loss_coeff (`float`, *optional*, defaults to 0.001):
+            The coefficient for the auxiliary load balancing loss.
+        moe_num_shared_experts (`int`, *optional*, defaults to 2):
+            The number of shared experts.
+        pad_token_id (`int`, *optional*, defaults to 2):
+            The padding token ID.
+    """
+
+    model_type = "aria_text_model"
+    base_config_key = "text_config"
+
+    def __init__(
+        self,
+        moe_intermediate_size: int = 4096,
+        moe_num_experts: int = 8,
+        moe_topk: int = 2,
+        moe_z_loss_coeff: float = 1e-5,
+        moe_aux_loss_coeff: float = 1e-3,
+        moe_num_shared_experts: int = 2,
+        pad_token_id=2,
+        **kwargs,
+    ):
+        super().__init__(pad_token_id=pad_token_id, **kwargs)
```

Comment:
```suggestion
        **super_kwargs,
    ):
        super().__init__(pad_token_id=pad_token_id, **super_kwargs)
```
Here using `super_kwargs` instead of `kwargs` will unpack the kwargs from the parent and fix the issue!

Reply:
Yes thank you, it does fix it!